pyiron_base/job/jobtype.py

```diff
@@ -95,8 +98,50 @@ def __new__(
             job.set_input_to_read_only()
         return job
 
+    @classmethod
+    def un_register_job_type(cls, job_name):
+        if job_name in cls._job_class_dict:
+            del cls._job_class_dict[job_name]
+        else:
+            raise KeyError(f"No JobType with name '{job_name}' found.")
+
+    @classmethod
+    def register_job_type(cls, cls_name, _cls):
```

Comment:
```suggestion
    def register(cls, job_class, job_name=None):
```
Since we don't allow custom class names, I would make it optional.

Reply:
The problem is a bit that it should be possible to register jobs on pure string basis like we do with the `JOB_CLASS_DICT`
```suggestion
    def register(cls, job_class_or_module_str, job_name=None):
```
I will try to make this as clear as possible